Source/WebCore/crypto/gcrypt/CryptoKeyOKPGCrypt.cpp

```diff
@@ -61,11 +67,11 @@ static bool supportedAlgorithmIdentifier(CryptoAlgorithmIdentifier keyIdentifier
 
 std::optional<CryptoKeyPair> CryptoKeyOKP::platformGeneratePair(CryptoAlgorithmIdentifier identifier, NamedCurve namedCurve, bool extractable, CryptoKeyUsageBitmap usages)
 {
-    if (namedCurve != NamedCurve::Ed25519)
+    if (namedCurve != NamedCurve::Ed25519 && namedCurve != NamedCurve::X25519)
         return { };
 
     PAL::GCrypt::Handle<gcry_sexp_t> genkeySexp;
-    gcry_error_t error = gcry_sexp_build(&genkeySexp, nullptr, "(genkey (ecdsa (curve \"Ed25519\") (flags eddsa)))");
+    gcry_error_t error = gcry_sexp_build(&genkeySexp, nullptr, namedCurve == CryptoKeyOKP::NamedCurve::Ed25519 ? "(genkey (ecdsa (curve Ed25519) (flags eddsa)))" : "(genkey (ecc (curve Curve25519)))");
```

Comment:
I think it would be cleaner, especially if Ed448 and X448 are supported at some point in the future, to disassociate the generation here into separate helper methods, already avoiding today's ternary operators.

For X25519, I would recommend employing the X25519 function on 32 bytes of random data, as per RFC7748, which would avoid these libgcrypt peculiarities regarding the leading byte.

All this can be discussed later and adjusted as necessary.

Reply:
I''ll do this refactoring in a follow up patch, thanks for the suggestion.